.github/workflows/integration_test.yml

```
@@ -30,11 +33,75 @@ jobs:
         uses: ./.github/actions/setup-complyctl
         # This is the test data for a specific product, and it may be updated in the future if necessary.
         with:
-          product: 'fedora'
-          catalog: 'cusp_fedora'
-          profile: 'fedora-cusp_fedora-default'
+          product: $PRODUCT
+          catalog: $CATALOG
+          profile: $PROFILE
 
-      # Test complyctl
-      - name: Test complyctl
+      - name: Test complyctl list
         run: |
-          complyctl list --plain
+          LIST_OUTPUT=$(complyctl list --plain)
+          echo $LIST_OUTPUT
+          echo $LIST_OUTPUT | grep -q "$CATALOG"
+
+      - name: Test complyctl plan
+        if: ${{ matrix.assessment-plan-status == 'original'}}
+        run: complyctl plan $CATALOG
+
+      - name: Test complyctl plan dry run
+        if: ${{ matrix.assessment-plan-status == 'customized'}}
+        run: complyctl plan $CATALOG --dry-run --out config_plan.yml
+
+      - name: Customize plan config
+        if: ${{ matrix.assessment-plan-status == 'customized'}}
+        run: |
+          cat << EOF > config.yml
+          frameworkId: cusp_fedora
+          includeControls:
+            - controlId: cusp_fedora_4-1
+              controlTitle: Account Protection
+              includeRules:
+                - accounts_umask_etc_login_defs
+              selectParameters:
+                - name: var_accounts_user_umask
+                  value: "027"
+          EOF
+
+      - name: Test complyctl plan with customized config
+        if: ${{ matrix.assessment-plan-status == 'customized'}}
+        run: complyctl plan $CATALOG --scope-config config.yml
+
+      # Validate OSCAL assessment-plan using go-oscal
+      - name: Validate assessment-plan
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # download go-oscal executable
+          gh release download --pattern "go-oscal_*_Linux_amd64" -R defenseunicorns/go-oscal
+          GO_OSCAL_EXECUTABLE=$(ls | grep go-oscal)
```

```suggestion
          GO_OSCAL_EXECUTABLE=$(ls go-oscal*)
```
I don't have a strong opinion, but using only ls without | and grep seems enough.